packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenPayment.sol

```diff
@@ -0,0 +1,286 @@
+        external
+        override
+    {
+        require(
+            hasAirnodeToPaymentDestinationSetterRoleOrIsManager(msg.sender),
+            "Not Airnode to payment destination setter"
+        );
+        require(
+            _paymentDestination != address(0),
+            "Invalid destination address"
+        );
+        airnodeToPaymentDestination[msg.sender] = _paymentDestination;
+        emit SetAirnodeToPaymentDestination(_paymentDestination, msg.sender);
+    }
+
+    /// @notice Make payments to gain access to Airnode
+    /// @dev In order for this function to be able to extend the whitelisting,
+    /// the Airnode operator must first grant the whitelist expiration extender
+    /// role to this contract, otherwise it will revert.
+    /// chainId-airnode-endpoint-requester pair gets authorized for the
+    /// duration specified. The amount to be paid is determined by the fee set
+    /// in the AirnodeFeeRegistry contract and the token price.
+    /// @param _chainId Id of the chain
+    /// @param _airnode Airnode address
+    /// @param _endpointId Id of the endpoint
+    /// @param _requesterAddress Requester address
+    /// @param _whitelistDuration Duration in seconds for which the requester
+    /// will be whitelisted
+    function makePayment(
+        uint256 _chainId,
+        address _airnode,
+        bytes32 _endpointId,
+        address _requesterAddress,
+        uint64 _whitelistDuration
+    ) external override {
+        require(_chainId != 0, "Invalid chainId");
+        require(_airnode != address(0), "Invalid Airnode address");
+        require(_requesterAddress != address(0), "Invalid requester address");
+        require(
+            _whitelistDuration != 0 &&
+                _whitelistDuration <= DEFAULT_MAXIMUM_WHITELIST_DURATION &&
+                _whitelistDuration >= getMinimumWhitelistDuration(_airnode),
+            "Invalid whitelist duration"
+        );
+
+        address requesterAuthorizerWithManager = IAirnodeRequesterAuthorizerRegistry(
+                airnodeRequesterAuthorizerRegistry
+            ).chainIdToRequesterAuthorizerWithManager(_chainId);
+        require(
+            requesterAuthorizerWithManager != address(0),
+            "No RequesterAuthorizersWithManager set for chainId"
+        );
+
+        (
+            uint64 expirationTimestamp,
+            uint192 indefiniteWhitelistCount
+        ) = IRequesterAuthorizerWithManager(requesterAuthorizerWithManager)
+                .airnodeToEndpointIdToRequesterToWhitelistStatus(
+                    _airnode,
+                    _endpointId,
+                    _requesterAddress
+                );
+        require(
+            indefiniteWhitelistCount == 0,
+            "Already whitelisted indefinitely"
+        );
+        require(
+            uint64(block.timestamp) + _whitelistDuration > expirationTimestamp,
+            "Already whitelisted"
+        );
```

Comment:
So just to be clear the way it is currently we require `now + whitelistduration > expirationTimestamp` i.e  if the user whitelisted for 3 months, we then attempt to whitelist for 1 month it would revert. The way it should work is if the user is whitelisted for 3 months and we attempt to whitelist for 1 month it would extend the whitelisting by 1 month to 4 months. Obviously we can extend only till the maximum limit after which it would revert.